src/plugins/discover/public/application/main/utils/adhoc_data_views.ts

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { DataView } from '@kbn/data-views-plugin/public';
```

Comment:
what we are doing here is adding another kind of state, that we set in our topnav and use in our SQL fetching 

let's sync at the start of the week, because there's 

https://github.com/elastic/kibana/pull/140169

which passes down the data view list down to `use_discover_state.ts`, and it wouldn't be difficult to pass it down to text base query fetching

it would be an opportunity to migrate the functionality in `adhoc_data_views.ts` to it's own hook like `use_data_views.ts`, where also 

https://github.com/elastic/kibana/blob/1990a98983eab55760f4e081d4cd716b282dee9b/src/plugins/discover/public/application/main/hooks/use_discover_state.ts#L201-L231

could be migrated to. anyway it's late and weekend already arrived. all other stuff is fixed which is great 🥳 . Let's sync about this start of next week. Have a nice one!

Reply:
`use_data_views.ts` added